### ISSUE TITLE
Yell at me about stuff

### ISSUE THREAD

**Issue by RainbowCookie32** [RainbowCookie32/rpcs3#1]
Don't actually yell at me. I'm creating this issue for talk about this build, instead of spamming the issue on RPCS3's repository

**Comment by RainbowCookie32** [comment]
Updated:
-MMJoystick support (hardcoded to L3, tested).
-Dualshock 4 support (hardcoded to L3, untested, should work).

https://ci.appveyor.com/api/buildjobs/uu7qwaq97grx0r97/artifacts/rpcs3-v0.0.6-7892-750849c4_win64.7z

**Comment by Marthgun** [comment]
does this work with xinput controllers?  can the button be re-mapped?

**Comment by RainbowCookie32** [comment]
> does this work with xinput controllers? can the button be re-mapped?

It works on XInput, MMJoystick and DualShock 4, hardcoded to the L3 button. Right now it can't be remapped, though I'm investigating how to allow remapping

**Comment by vsub** [comment]
I don't know if I am doing something wrong but I find two problems

1.At least here with my gamepad,MMJoystick is not working
2.Switching speed in any game will run the next game at the last fps limit ignoring the Frame Limit settings.
Lets say I run a game that is running at 60fps,then I press L3 to switch to 30fps and then run another game(or the same)that is also running at 60fps but in this case,the game will run at 30fps until you press L3 again or restart rpcs3
It's like this feature is setting some global variable that is overwriting the one set by the default one.

**Comment by RainbowCookie32** [comment]
I'll check it out when I get home in a few hours and I'll see what I can do

**Comment by RainbowCookie32** [comment]
I just checked with my controller and it works, it's mapped to Button 11, which should be L3 but might be different on your controller? I did notice what you said on your second point and I'll have a look now and see what I can do.
@Chief5193 It'd be more useful if you said what's your problem instead of just adding a thumbs down emoji. Or at least explain why you add it

**Comment by vsub** [comment]
On my gamepad which support XInput\DInput L3 is Button9
It is probably Button11(are you sure)on yours because your L2 and R2 are buttons not triggers\analogs

I also have another gamepad that is  DInput only and the L3 is Button10

I don't know this coding language but is every button set to a variable?(it is supposed to be because rpcs3 is reading the "Default Profile.yml" so the buttons will be set the way you want not the way they are supposed to be)
Maybe this will give you some ideas

```
 std::array<u16, XInputKeyCodes::KeyCodeCount> button_values = GetButtonValues(state);
 			if(button_values[**the variable representing LS or maybe the PS Button**) {
 				if(!prev_rs_state[padnum]) {
 					DuranteFrameLimiter::get().toggleLimit();
 					prev_rs_state[padnum] = true;
 					//LOG_ERROR(HLE, "TOGGLE");
 }
```

This way that function will not be hardcoded

**Comment by RainbowCookie32** [comment]
Yeah I'm looking at how the mapping system works and I'll try to add it to the Pad settings, seems like buttons are a bit random on MMJoystick devices. I "fixed" the persistent limiter btw, I'll push the changes now

**Comment by RainbowCookie32** [comment]
Forgot that I had appveyor building 😅 
Updated build, discards the framelimiter values when emulation is stopped: https://ci.appveyor.com/api/buildjobs/2c85q3vxs4150nnf/artifacts/rpcs3-v0.0.6-7893-1de981b8_win64.7z
Working on remapping

**Comment by vsub** [comment]
Thanks.
When you are done with the remapping,can you use the latest version when compiling(0.0.6-7901 or maybe newer)

**Comment by RainbowCookie32** [comment]
Yeah sure

**Comment by RainbowCookie32** [comment]
Updated to master, still working on remapping. I think I'm making progress.
[Link](https://ci.appveyor.com/api/buildjobs/gt2wvs2he508vk8b/artifacts/rpcs3-v0.0.6-7912-72ae9bea_win64.7z)

**Comment by RainbowCookie32** [comment]
Nailed it, remapping is working on MMJoystick. I just needed my coffee 😄

**Comment by RainbowCookie32** [comment]
New Update!
Remapping on XInput and MMJoystick devices. Get it [here](https://ci.appveyor.com/api/buildjobs/2u9hpfoghpv5nofq/artifacts/rpcs3-v0.0.6-7913-f3965d81_win64.7z)!
DS3/4 are probably stuck on L3 though, since I think RPCS3 doesn't allow remapping of those controllers

**Comment by vsub** [comment]
Thanks,it works now in XInput too and I can set the key in both handlers
The only problem is that in MMJoyStick I cannot set my home button...in MMJoyStick rpcs3 don't recognize in(it's the same thing on master but that probably just how MMJoyStick works)

Btw one stupid question....why the rev number is 0.0.6-7913 if master is 7901

**Comment by RainbowCookie32** [comment]
The 7xxx number is the number of commits on the repository of I recall
correctly, so that's why it's different from master. Don't know what's up
with MMJoystick not letting you set up the home button, but if it happens
in master as well it might be a limitation of the handler, as you said.
Good to know that everything works as intended :D

**Comment by RainbowCookie32** [comment]
Updated to latest master https://ci.appveyor.com/api/buildjobs/5y6bd2n27jpu2lfr/artifacts/rpcs3-v0.0.6-7921-a02dca0b_win64.7z

**Comment by vsub** [comment]
One simple question
If the version number is not the same as the master,how to know on which rev the link above for example is based on
If there is no way to tell,can you post the version number on which the link you post(for future updates)is based on

**Comment by RainbowCookie32** [comment]
You can check on the commits list. Last one would be mine (Merge pull request * from RPCS3/master) so ignore that one. The one before that one is the last commit that was on master when I updated. But I'll update the comments now with the versions and try to include it in future ones

**Comment by RainbowCookie32** [comment]
Updated to master (0.0.6-7909) https://ci.appveyor.com/api/buildjobs/w9r5wme52wijvbbh/artifacts/rpcs3-v0.0.6-7924-f7584da3_win64.7z

**Comment by RainbowCookie32** [comment]
Updating to master (0.0.6-7932) https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/23555033/artifacts

**Comment by Rotockter** [comment]
Sadly not working for me with XInput nor Ds4. Its stays at whatever framelimit option i set in the game config.

**Comment by RainbowCookie32** [comment]
You need to use Auto framelimiter for it to work iirc

**Comment by vsub** [comment]
Anything other than 30 actually and the game must support running at above 30fps and you hardware to be powerful enough to give you above 30fps
Persona 5 for example without the 60fps patch will run at 30fps max no matter what.

**Comment by Rotockter** [comment]
Me: What is Vulkan ?....google vulkan.....OoooKkkk next Open GL api ahmmm.....

Then i set it to auto or off the FPS is all over the place. In some rooms like 120 fps and in cutscenes. In the citiy in the beginning between 25-42 fps. In battles like 60 fps.
Also some areas are affected if i have full screen or not on. like in the first dungeon. No full screen like 120 fps with fullscreen like 60 fps.

When i set it to 30 fps its all stable but the battle is a bit slugish. And the Audio is way more crackling.
When i set it to 60 fps its like auto and off sometimes.

So i need this Vulkan stuff. Can that simply installed or do i need a compatible grafic card ?

Also Thanks for the fast reply.

**Comment by AtelierFox** [comment]
If this is now working, with remapping too, would it be reasonable to set this fork as a pull request into the original rpcs3 repo? While it is technically a hack, an FPS toggle is a useful debugging feature.

**Comment by RainbowCookie32** [comment]
I do agree with you, but with my limited knowledge in this area it does
look like an ugly hack and I really doubt it'd be accepted

**Comment by vsub** [comment]
Hi,can you update the build to the latest version.
And btw,it's good the fps problem to be fixed but right now,I am actully using the bug as a feature because of the hotkey
When I want to reach a destination faster,I just switch to 60fps to run twice as fast in tox1\2

**Comment by RainbowCookie32** [comment]
Currently building [here](https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/23818957/artifacts). Updated to 0.0.6-7972

**Comment by Shadow5YA** [comment]
Is there any way to get the FPS limiter to work with OpenGL as well? Something about the Vulkan API doesn't set the FPS probably, so battles in games like Graces F run quicker than they normally do even at the intended 60fps, but slower than intended when using the limiter to set it to 30.

If I use openGL, battles run normally at 60fps, but then the FPS limiter doesn't work on openGL to use for the 30fps cutscenes.

**Comment by RainbowCookie32** [comment]
I'll take a look at OpenGL and see if I can implement it there as well. I
will probably be able to work on it next week

**Comment by RainbowCookie32** [comment]
After way too long, I updated it https://ci.appveyor.com/api/buildjobs/ni23qsnh1kojyq7q/artifacts/rpcs3-v0.0.6-8272-6fc68ae1_win64.7z
If someone's still around, feel free to test 😉

**Comment by vsub** [comment]
Just as info,I clear both games ToX 1 and 2 with that feature and I did not encounter a single crash

**Comment by HighlowEnough** [comment]
Tales of Graces F
It doesn't work when you run OpenGL No matter how hard you press the L3 button, the frame does not change.

Volkan it works very well
But even with 60FPS, the battle speed is too fast to play.

Tales of Xillia 2

I was able to play properly without colliding.
Except for a few minor problems, the play is perfect.

(config.yml ppu thread 2 to 1 BGM fixes
Side effects slow down in the battle of some monsters and the final boss.)


The only problem is the DLC.

Thank you :) RainbowCookie32

**Comment by RainbowCookie32** [comment]
Updated the repo to 8313, currently building [here](https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/25628557/artifacts)

@HighlowEnough Yep, the hack is only implemented in Vulkan, so nothing will happen in OpenGL. Glad this could help you :D

**Comment by HighlowEnough** [comment]
Thank you for your kind explanation.

The new build shows 70% Brightness.

But that's all right. You can play with your old build or just play.
Most of all, it's great to be able to play games for hours without random crashes.

Thank you so much just for updating the build. :)

**Comment by RainbowCookie32** [comment]
I'll update it again when kd's PR is merged, maybe it fixes it?

**Comment by vsub** [comment]
> Most of all, it's great to be able to play games for hours without random crashes.

I don't think the changes done here have anything to do with the random crashes
It is most likely the PPU Thread set to 1 that makes the game more stable

**Comment by klepp0906** [comment]
thank you for this <3

**Comment by RainbowCookie32** [comment]
Updated the repo to 0.0.6-8332, building here https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/25809390/artifacts

**Comment by xXDeonoXx** [comment]
> Updated the repo to 0.0.6-8332, building here https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/25809390/artifacts

Can't make the change between 30 and 60 fps on that version, using a original xbox one controller, vulkan api and the same settings from a previous version from this post.

**Comment by vsub** [comment]
@RainbowCookie32 are you sure this is a build with the brightness and fps feature
The rev looks like the normal version(except the warning),no brightness fix,no fps switching and no extra button in the Input settings.

**Comment by RainbowCookie32** [comment]
Good catch guys, I messed up there. I was updating the wrong branch

**Comment by RainbowCookie32** [comment]
Here we go https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/25939952/artifacts
Updated to latest, sorry for the brainfart

**Comment by RainbowCookie32** [comment]
Update to 0.0.6-8400 currently building [here](https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/26127266/artifacts)
99% sure that I didn't mess up this time

**Comment by 3litkhongsay** [comment]
> 
> 
> Update to 0.0.6-8400 currently building [here](https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/26127266/artifacts)
> 99% sure that I didn't mess up this time
Thanks bro ! 
I used the 0.0.6-8400 and 0.0.6-8300 versions to get the wrong color when playing Tales of graces, but 8180 did not, please help me.

**Comment by 3litkhongsay** [comment]
![3131](https://user-images.githubusercontent.com/53133340/61586788-4f029180-aba6-11e9-9a43-8eb7a0dfb039.PNG)

**Comment by RainbowCookie32** [comment]
If it also happens on master (which is probably the case), then it's not caused by this hack and it was introduced by a change in the emulator. I'm not really qualified to fix it. Try searching for when the regression happened with normal RPCS3 builds and open an issue on the main repository (if there's not one open already) so it can be fixed

**Comment by 3litkhongsay** [comment]
> 
> 
> If it also happens on master (which is probably the case), then it's not caused by this hack and it was introduced by a change in the emulator. I'm not really qualified to fix it. Try searching for when the regression happened with normal RPCS3 builds and open an issue on the main repository (if there's not one open already) so it can be fixed

Please help me update 8180, I am very thankful 
https://drive.google.com/open?id=1CL1ySUq4SCXD5N3JhclOjWBdJpWm6U2T
!

**Comment by RainbowCookie32** [comment]
Updated, currently building https://ci.appveyor.com/project/RainbowCookie/rpcs3/build/artifacts
Should work fine, haven't tested it myself though

**Comment by dawnbomb** [comment]
Indeed, L3 (the point of the build) Doesn't work.
That said, can we change it to R3 please, R3 doesn't do anything in battle, but L3 changes Elize's form, and for some characters do a taunt and get meter if they equipped the skill for it. also, people with bad controllers may press L3 on accident when walking around.

**Comment by RainbowCookie32** [comment]
Yeah, compilation for that branch failed miserably and I kind of broke my local branch a bit. I'll try to fix it today

**Comment by RainbowCookie32** [comment]
Alright, it's building now here https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/26920879/artifacts
This one should work, it compiled fine locally

**Comment by HighlowEnough** [comment]
Press the pad analog stick L3 button and now both 30FPS and 60FPS work properly.
RainbowCookie32 Thank you.

**Comment by HernanRC2** [comment]
> 
> 
> Alright, it's building now here https://ci.appveyor.com/project/RainbowCookie/rpcs3/builds/26920879/artifacts
> This one should work, it compiled fine locally

Thanks. I downloaded this build , 30FPS-60FPS works fine, but 70% brightness problem is not fixed.

**Comment by MystesofEternity** [comment]
Thank you very much for all the hard work @RainbowCookie32 !
